server/reload.go

```diff
@@ -705,6 +705,46 @@ func (s *Server) Reload() error {
 	if err := s.reloadOptions(curOpts, newOpts); err != nil {
 		return err
 	}
+
+	s.mu.Lock()
+	disconnectClients := []*client{}
+	checkClients := func(kind int, clients map[uint64]*client, set PinnedCertSet) {
+		for _, c := range clients {
+			if c.kind == kind && !c.matchesPinnedCert(set) {
+				disconnectClients = append(disconnectClients, c)
+			}
+		}
+	}
+	if !reflect.DeepEqual(newOpts.TLSPinnedCerts, curOpts.TLSPinnedCerts) {
+		checkClients(CLIENT, s.clients, newOpts.TLSPinnedCerts)
+	}
+	if !reflect.DeepEqual(newOpts.MQTT.TLSPinnedCerts, curOpts.MQTT.TLSPinnedCerts) {
+		checkClients(MQTT, s.clients, newOpts.MQTT.TLSPinnedCerts)
```

Comment:
That won't work... MQTT and Websocket clients are still c.kind == CLIENT. For c.kind == CLIENT, then you could either check for isWebsocket() or isMqtt(), or invoke c.clientType(). But again, only when c.kind == CLIENT.